web/src/app/util/calendar/CalendarEventWrapper.tsx

```diff
@@ -53,6 +55,10 @@ export default function ScheduleCalendarEventWrapper({
   const [showEditDialog, setShowEditDialog] = useState('')
   const [showDeleteDialog, setShowDeleteDialog] = useState('')
 
+  // for showing "Visit Schedule" buttons when on user profile
+  const [path] = useLocation()
+  const onProfile = path.includes('users') || path.includes('profile')
```

Comment:
Can we do this through a prop like `showScheduleLink` rather than relying on magic?

Reply:
what is magical about this

Reply:
This component is called like this:

```
components={{
            // @ts-expect-error Property 'children' does not exist on type - yes it does
            eventWrapper: CalendarEventWrapper,
            toolbar: () => null,
          }}
```

Where recently you removed it being called as `eventWrapper: ({ children }) => (<CalendarEventWrapper />)` or something similar so it wouldn't render multiple times.

Reply:
When I refer to the approach as "magical," I'm emphasizing the indirect nature of rendering based on the URL path, rather than through explicit props. The core of my concern is about maintainability and clarity of the component's behavior.

Using explicit props like showScheduleLink for rendering decisions aligns better with best practices in component design. It ensures that the component's behavior is directly controlled by its parent, making it more predictable and easier to manage. This approach is especially beneficial in scenarios like routing changes, URL modifications, and testing. If we rely on the URL path, it introduces an additional layer of complexity and fragility. Changes in routing or URL structure would directly impact the component, potentially leading to unforeseen issues.

In essence, by using props, we're maintaining a clear contract between the parent and the child component. It makes the data flow and rendering logic more transparent and cohesive. This not only aids in current understanding and maintenance but also simplifies any future modifications, whether it's changing routes, refactoring, or writing tests.

Reply:
https://github.com/target/goalert/commit/57a61e89af988a578d09b58d0e962c1bb09cf945#diff-aa1aba9c2ce6c7a6f0e6b6771d34bdec24c2c56b1be30a0e264712cebbd22d4aL387

Reply:
Swap the component, rather than render inline:
```
            eventWrapper: props.showScheduleLink
              ? CalendarEventWrapperWithLink
              : CalendarEventWrapper,
```

90% of the code in `CalendarEventWrapper` is about temp schedules and overrides, so rather than adding more complexity to it, we could just make a simple one for the new event type.

Alternatively, you could do something like this:
```

function CalendarEventWrapperWithLink(
  props: ScheduleCalendarEventWrapperProps,
): JSX.Element {
  return <CalendarEventWrapper {...props} showScheduleLink />
}
```

However, `CalendarEventWrapper` will still need all the conditionals throughout since it's being used in two different ways.

Either option will solve the problem without re-introducing the render bug.

Reply:
So... Swap one complexity for the other. I'll make a wrapper component, sure.

Reply:
Done. Resolving.